lib/src/command/dependency_services.dart

```diff
@@ -362,17 +362,31 @@ class DependencyServicesApplyCommand extends PubCommand {
       }
     }
 
-    if (pubspecEditor.edits.isNotEmpty) {
-      writeTextFile(entrypoint.pubspecPath, pubspecEditor.toString());
-    }
-    if (lockFileEditor != null && lockFileEditor.edits.isNotEmpty) {
-      writeTextFile(entrypoint.lockFilePath, lockFileEditor.toString());
-    }
+    final updatedLockfile = lockFileEditor == null
+        ? null
+        : LockFile.parse(lockFileEditor.toString(), cache.sources);
```

Comment:
Since JSON is a strict subset of YAML, you don't have to use `YamlEditor` for modifying the `pubspec.lock` anymore...

Reply:
Yeah - it doesn't really hurt either. Leaving this for now